### PR TITLE
Fix product tests previously migrated to native fs to not use legacy fs

### PR DESCRIPTION
**Pull request by anusudarsan**
## Description

Tests that were migrated to native fs was still using hdfs. So migrating them completely to use native-fs

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Comment by anusudarsan** [comment]
@ebyhr can you trigger ci with secrets please?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlluxioCaching.java [hunk 1]
```diff
@@ -84,6 +94,8 @@ private void testReadFromTable(String tableNameSuffix)
     private void createTestTable(String tableName)
     {
         onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery("DROP SCHEMA IF EXISTS delta.test_caching");
```
Comment (nineinchnick) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlluxioCaching.java:97]: Would be nice to make sure the schema here is the same as the one used in `tableName`.

**Review by nineinchnick — APPROVED** [review]

**Comment by ebyhr** [comment]
/test-with-secrets sha=62892148a69e36791a920375a7686a539f5e601a

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestHiveAndDeltaLakeCompatibility.java [hunk 2]
```diff
@@ -59,14 +69,22 @@ public void testInformationSchemaColumnsOnPresenceOfHiveView()
...
-                .hasMessageContaining("not a Delta Lake table");
+            onTrino().executeQuery("CREATE TABLE " + hiveTableName + " AS SELECT 1 a");
 
-        onTrino().executeQuery("DROP TABLE " + hiveTableName);
+            assertThatThrownBy(() -> onTrino().executeQuery(format("CALL delta.system.unregister_table('%s', '%s')", schemaName, baseTableName)))
+                    .hasMessageContaining("not a Delta Lake table");
+        }
+        finally {
+            onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+            onTrino().executeQuery("DROP SCHEMA " + schemaName);
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestHiveAndDeltaLakeCompatibility.java:87]: This line may throw an exception if `CREATE SCHEMA` failed. How about adding `IF EXISTS` option or moving `CREATE SCHEMA` before `try` block?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestHiveAndDeltaLakeCompatibility.java [hunk 3]
```diff
@@ -59,14 +69,22 @@ public void testInformationSchemaColumnsOnPresenceOfHiveView()
...
     {
+        String schemaName = "test_unregister_not_delta_lake_schema_" + randomNameSuffix();
+
         String baseTableName = "test_unregister_not_delta_table_" + randomNameSuffix();
-        String hiveTableName = "hive.default." + baseTableName;
+        String hiveTableName = format("hive.%s.%s", schemaName, baseTableName);
 
-        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " AS SELECT 1 a");
+        try {
+            onTrino().executeQuery("CREATE SCHEMA IF NOT EXISTS hive." + schemaName + " with (location = 's3://" + bucketName + "/test_unregister_to_delta')");
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestHiveAndDeltaLakeCompatibility.java:78]: nit: Uppercase `with`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestHiveAndDeltaLakeCompatibility.java [hunk 4]
```diff
@@ -34,12 +36,20 @@
...
+        bucketName = requireNonNull(System.getenv("S3_BUCKET"), "Environment variable not set: S3_BUCKET");
+    }
+
     @Test(groups = {DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
     public void testInformationSchemaColumnsOnPresenceOfHiveView()
     {
         // use dedicated schema so we control the number and shape of tables
         String schemaName = "test_redirect_to_delta_information_schema_columns_schema_" + randomNameSuffix();
-        onTrino().executeQuery("CREATE SCHEMA IF NOT EXISTS hive." + schemaName);
+        onTrino().executeQuery("CREATE SCHEMA IF NOT EXISTS hive." + schemaName + " with (location = 's3://" + bucketName + "/test_redirect_to_delta')");
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestHiveAndDeltaLakeCompatibility.java:52]: nit: Uppercase `with`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlluxioCaching.java [hunk 5]
```diff
@@ -81,9 +92,11 @@ private void testReadFromTable(String tableNameSuffix)
     /**
      * Creates a table which should contain around 6 2 MB parquet files
      */
-    private void createTestTable(String tableName)
+    private void createTestTable(String tableName, String schemaName)
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlluxioCaching.java:95]: I expect `tableName` isn't a qualified name from these variable names. Also, we shouldn't create new schemas per table. Can we move the schema creation logic to caller side? 

```suggestion
    private void createTestTable(String schemaName, String tableName)
```

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlluxioCaching.java [hunk 6]
```diff
@@ -81,9 +92,11 @@ private void testReadFromTable(String tableNameSuffix)
...
     {
         onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery("DROP SCHEMA IF EXISTS " + schemaName);
+        onTrino().executeQuery("CREATE SCHEMA " + schemaName + " with (location = 's3://" + bucketName + "/test_delta_cached')");
```
Comment (ebyhr) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlluxioCaching.java:99]: nit: Uppercase `with`.

**Review by ebyhr — APPROVED** [review]

**Comment by anusudarsan** [comment]
@ebyhr addressed comments

**Pull request merged**